File: Cryptography and security libraries and algorithms/C/Replit Ghostwriter/Timing attacks.c
#include <openssl/crypto.h>

const char password[] = "password";
const char user_input[] = "user-input";
const int password_len = strlen(password);
const int user_input_len = strlen(user_input);
// Log the user in if the password and user input match
int main() {
  if (strcmp(crypt(password, user_input), user_input) == 0) {
    printf("You are logged in.\n");
  } else {
    printf("You are not logged in.\n");
  }
  return 0;
}

File: Cryptography and security libraries and algorithms/Python/Replit Ghostwriter/Timing attacks.py
import hmac

password = b'password'
user_input = b'user-input'

# Log the user in if the password and user input match
if hmac.new(password, user_input, hashlib.sha256).hexdigest() == '0a5b0b8d9f3b6b3b9b9b9b9b9b9b9b9b9b9b9b9b9b9b9b9b' :
    print('You are logged in')
else:
    print('You are not logged in')